src/tracks/ui/EditCursorOverlay.cpp

```diff
@@ -98,34 +98,11 @@ void EditCursorOverlay::Draw(OverlayPanel &panel, wxDC &dc)
    if (!onScreen)
       return;
 
-   auto &trackPanel = TrackPanel::Get( *mProject );
-   if (auto tp = dynamic_cast<TrackPanel*>(&panel)) {
-      wxASSERT(mIsMaster);
-      AColor::CursorColor(&dc);
-
-      // Draw cursor in all selected tracks
-      tp->VisitCells( [&]( const wxRect &rect, TrackPanelCell &cell ) {
-         const auto pTrackView = dynamic_cast<TrackView*>(&cell);
-         if (!pTrackView)
-            return;
-         const auto pTrack = pTrackView->FindTrack();
-         if (pTrack->GetSelected() ||
-             TrackFocus::Get( *mProject ).IsFocused( pTrack.get() ))
-         {
-            // AColor::Line includes both endpoints so use GetBottom()
-            AColor::Line(dc, mLastCursorX, rect.GetTop(), mLastCursorX, rect.GetBottom());
-            // ^^^ The whole point of this routine.
-
-         }
-      } );
-   }
-   else if (auto ruler = dynamic_cast<AdornedRulerPanel*>(&panel)) {
```

Comment:
Another point of the old implementation of the zero-width cursor is that the function DrawOverlays() knows how to redraw certain simple strokes over the track panel area cheaply, without repainting everything, and existing calls to DrawOverlays() after change of selection assume that the erasing of old selection and drawing of new will be done.

Reply:
And that happens only for drawing implemented by subclasses of Overlay.  Not TrackPanelDrawable.

Reply:
As far as i can see, changing selection involves redrawing anyway (i didn't add any line to invoke redrawing), so from that perspective i don't see difference. Right now i don't see solid solution for solving this issue with overlays, i can come up only with workarounds.

Reply:
It may be a problem that we do too much unnecessary redrawing and should figure out how to do less of it.

I don't see why you can't have the EditCursorOverlay dynamic_cast to tack affordance cell types and then that's not null, then draw over the cell, just as with TrackView.

Reply:
Then the only reason for having a class for affordance cell (which is required to be a CommonTrackCell now) would be to detect it's type with dynamic_cast (or to check for WaveTrackAffordanceControls and NoteTrackAffordanceControls separately). But when drawing is done in ChannelGroup, we don't need neither of these (separate class, type check).

Reply:
This is not sufficient reason not to make the affordance cell class.

When I rebase my long branch onto the part of affordances already merged, and resolved conflicts, I needed the new common class for other reasons.

The code in TrackPanel may be simpler, avoiding casts, but that also does not address the possible performance problems.

Short answer:  I still have my doubts about moving the drawing.  It may prove to be better this way.  But, I also don't want to delay merging this, and the appearance is right.

Therefore:  don't delete the lines above, but comment them out, as the possibly useful alternative way of drawing the cursor, which we may want to use again in future.